hack/prep-cluster-for-mcd-push.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+set -euo pipefail
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+# shellcheck disable=SC1091
+source "$SCRIPT_DIR/hacklib"
+
+copy_mcd_to_disk() {
+  pod="$1"
+
+  oc debug \
+    -n "$MCO_NAMESPACE" \
+    -c "$MCD_CONTAINER_NAME" \
+    --one-container=true \
+    "pod/$pod" -- cp -v /usr/bin/machine-config-daemon "$ROOTFS_MCD_PATH"
+}
+
+main() {
+  # Disable the cluster version operator
+  # --replicas 1 out-of-the-box
+  oc scale --replicas 0 -n openshift-cluster-version deployments/cluster-version-operator
+
+  # Disable the MCO
+  # --replicas 1 out-of-the-box
+  oc scale --replicas 0 -n "$MCO_NAMESPACE" deployments/machine-config-operator
+
+  # Copy the MCD from the container to rootfs first to avoid startup issues
+  echo "Copy MCD from container to /rootfs"
+  mcd_pods="$(oc get pods -n "$MCO_NAMESPACE" -l 'k8s-app=machine-config-daemon' -o go-template='{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}')"
+  for pod in $mcd_pods; do
+    copy_mcd_to_disk "$pod" &
+  done
+
+  echo "Waiting for copies to finish..."
+  wait
+  echo "Done"
+
+  # Now we modify the MCD daemonset to copy the MCD binary from rootfs and run it
+  read -r -d '' patch << EOM
+spec:
+  template:
+    spec:
+      containers:
+      - args:
+          - -c
+          - "sha256sum $ROOTFS_MCD_PATH && cp $ROOTFS_MCD_PATH /usr/local/bin/machine-config-daemon && /usr/local/bin/machine-config-daemon start -v 4"
```

Comment:
I'm curious, why have you taken the approach of copying the binary rather than doing something more similar to `./hack/cluster-push.sh` but only pushing the new image for the MCD?

Related wondering: why do we lump all of our binaries into one image?

Reply:
I did it this way because it's a bit faster to build a single binary and push the binary instead of that approach. That's not to say that `./hack/cluster-push.sh` isn't valuable 😄 .

Reply:
I just remembered why I wrote these. I was trying to have a scripted way of doing the following: https://github.com/openshift/machine-config-operator/blob/master/docs/HACKING.md#developing-the-mcd-without-building-images. I'll update the docs to point to the scripts and explain how they work, etc.

Reply:
Yep makes sense! I was kinda thinking the increase in speed had to do with not changing operator/controller/server, but I guess building and pushing a container is slow